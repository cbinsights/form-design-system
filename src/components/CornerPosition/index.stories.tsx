import React from 'react';
import CheckIcon from 'icons/react/CheckIcon';
import Badge from 'components/Badge';
import CornerPosition from '.';

const Template = (args) => <CornerPosition {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  decoration: <Badge label="42" />,
  children: <button>lorem ipsum</button>,
};

export const Variations = (): JSX.Element => (
  <>
    <CornerPosition decoration={<Badge label="alert" />}>
      <div
        style={{
          border: '1px dashed #f09',
          padding: '5px',
        }}
      >
        CornerPosition can wrap <b>anything</b>!
      </div>
    </CornerPosition>
    <br />
    <br />
    <CornerPosition decoration={<Badge label={<CheckIcon size="xs" color="#fff" />} />}>
      <img src="https://placeimg.com/80/81/animals" />
    </CornerPosition>
  </>
);

export default {
  component: CornerPosition,
  title: 'components/CornerPosition',
  parameters: {
    componentSubtitle:
      'Absolutely positions a decoration node to the top right of another node.',
    docs: {
      fds: {
        related: ['Badge'],
      },
    },
  },
};
