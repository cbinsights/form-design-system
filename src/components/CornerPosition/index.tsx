import React from 'react';

export interface CornerPositionProps {
  /** Node to decorate */
  children: React.ReactNode;
  /** Decoration to add to Node */
  decoration: React.ReactNode;
}

const CornerPosition = ({ decoration, children }: CornerPositionProps): JSX.Element => (
  <div className="cornerPosition-root">
    {children}
    {decoration && <div className="cornerPosition-decoration">{decoration}</div>}
  </div>
);

export default CornerPosition;
