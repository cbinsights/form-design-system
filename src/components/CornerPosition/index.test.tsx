import React from 'react';
import { shallow } from 'enzyme';

import CornerPosition from '.';

describe('CornerPosition component', () => {
  it('matches snapshot', () => {
    const component = shallow(
      <CornerPosition decoration={<span />}>
        <div />
      </CornerPosition>
    );
    expect(component).toMatchSnapshot();
  });
});
