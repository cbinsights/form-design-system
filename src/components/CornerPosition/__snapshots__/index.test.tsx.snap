// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CornerPosition component matches snapshot 1`] = `
<div
  className="cornerPosition-root"
>
  <div />
  <div
    className="cornerPosition-decoration"
  >
    <span />
  </div>
</div>
`;
